java/org/apache/coyote/Request.java

```diff
@@ -852,25 +854,17 @@ public boolean isProcessing() {
      *
      * @param contentType a content type header
      */
-    private static String getCharsetFromContentType(String contentType) {
+    private static String getCharsetFromContentType(String contentType) throws IOException {
```

Comment:
The PR won't compile because of this addition.  Look at how other code (e.g. in Response) handles the possibility of an IOException. Hint: the method signature should be unchanged from the original